src/main/java/io/cdap/plugin/salesforce/plugin/source/streaming/SalesforcePushTopicListener.java

```diff
@@ -129,20 +128,103 @@ protected void customize(Request exchange) {
 
     // Now set up the Bayeux client itself
     BayeuxClient client = new BayeuxClient(oAuthInfo.getInstanceURL() + DEFAULT_PUSH_ENDPOINT, transport);
-    client.handshake();
-
     return client;
   }
 
-  private void waitForHandshake(BayeuxClient client,
-                                long timeoutInMilliseconds, long intervalInMilliseconds) {
+  public void createSalesforceListener() throws Exception {
+    bayeuxClient = getClient(credentials);
+    bayeuxClient.getChannel(Channel.META_HANDSHAKE).addListener
+      ((ClientSessionChannel.MessageListener) (channel, message) -> {
+
+        boolean success = message.isSuccessful();
+        if (!success) {
+          String error = (String) message.get("error");
+          if (error != null) {
+            LOG.error("Error during HANDSHAKE: {}", error);
+            throw new RuntimeException("Error in meta handshake , errorMessage:" + error);
+          }
+          if (message.get("exception") instanceof Exception) {
+            Exception exception = (Exception) message.get("exception");
+            if (exception != null) {
+              LOG.error("Exception during HANDSHAKE", exception);
+              throw new RuntimeException("Error in meta handshake ", exception);
+            }
+          }
```

Comment:
Is it possible that `!message.isSuccessful()` and message doesn't contain `error` or `exception`?

Reply:
That is unlikely to happen. We haven't come across any message that is not successful and does not contain any `error` or `exception`

All messages having `"successful": false` contain the `error` key.

Some common Streaming API Error Messages: https://developer.salesforce.com/docs/atlas.en-us.api_streaming.meta/api_streaming/streaming_handling_errors.htm